sdk/src/rustls/temp_signer.rs

```diff
@@ -0,0 +1,468 @@
+#![allow(dead_code)]
+
+//! Temporary signing instances for testing purposes.
+//!
+//! This module contains functions to create self-signed certificates
+//! and provision [`RustlsSigner`] instances for each of the supported signature
+//! formats.
+//!
+//! Private-key and signing certificate pairs are created in a directory
+//! provided by the caller. It is recommended to use a temporary directory
+//! that is deleted upon completion of the test. (We recommend using
+//! the [tempfile](https://crates.io/crates/tempfile) crate.)
+//!
+//! This module should be used only for testing purposes.
+
+// Since this module is intended for testing purposes, all of
+// its functions are allowed to panic.
+#![allow(clippy::panic)]
+#![allow(clippy::unwrap_used)]
+
+use std::{
+    io::Write,
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+};
+
+use crate::{rustls::RustlsSigner, signer::ConfigurableSigner, signer::Signer};
+
+/// Create a signer that can be used for testing purposes.
+///
+/// Can generate a [`Signer`] instance for all supported formats.
+///
+/// # Arguments
+///
+/// * `path` - A directory (which must already exist) to receive the temporary
+///   private key / certificate pair.
+/// * `alg` - A format for signing. Must be one of (`rs256`, `rs384`, `rs512`,
+///   `ps256`, `ps384`, `ps512`, `es256`, `es384`, `es512`, or `ed25519`).
+/// * `tsa_url` - Optional URL for a timestamp authority.
+///
+/// # Returns
+///
+/// Returns a tuple of `(signer, sign_cert_path)` where `signer` is
+/// the [`Signer`] instance and `sign_cert_path` is the path to the
+/// signing certificate.
+///
+/// # Panics
+///
+/// Can panic if unable to invoke OpenSSL executable properly.
+pub fn get_temp_signer<P: AsRef<Path>>(path: P) -> (RustlsSigner, PathBuf) {
+    let (sign_cert_path, pem_key_path) = make_key_path_pair(path, "temp_key");
+    //
+    create_x509_key_pair(
+        &sign_cert_path,
+        &pem_key_path,
+        false,
+        Some("rsa_padding_mode:pss"),
+        Some("-sha256"),
+    );
+
+    (
+        RustlsSigner::from_files(&sign_cert_path, &pem_key_path, "ps256".to_string(), None)
+            .unwrap(),
+        sign_cert_path,
+    )
+}
+
+fn make_key_path_pair<P: AsRef<Path>>(path: P, key_name: &str) -> (PathBuf, PathBuf) {
+    let mut sign_cert_path = path.as_ref().to_path_buf();
+
+    let mut pem_key_path = sign_cert_path.join(key_name);
+    pem_key_path.set_extension("pem");
+
+    sign_cert_path.push(key_name);
+    sign_cert_path.set_extension("pub");
+
+    (sign_cert_path, pem_key_path)
+}
+
+// The .x509 directory at the root of this repo is flagged
+// as outside of source control via .gitignore.
+//
+// This function panics if unable to invoke openssl as expected.
+//
+fn create_x509_key_pair(
+    sign_cert_path: &Path,
+    pem_key_path: &Path,
+    has_priv_key: bool,
+    rsa_padding_mode: Option<&str>,
+    sha_mode: Option<&str>,
+) {
+    let mut openssl = Command::new("openssl");
```

Comment:
I took care of the openssl uses in the Rust code, but and did not change these calls to it.